src/modules/direction/components/MaterialsContainer.tsx

```diff
@@ -25,49 +22,59 @@ const MaterialsContainer: React.FC<IMaterialsContainerProps> = ({
 }) => {
   const classes = useStyles();
 
-  const { posts, meta }  = useSelector(
+  const {
+    posts,
+    meta: { loading, isLastPage, pageNumber },
+  } = useSelector(
     (state: RootStateType) => state.directions[direction.name].materials,
   );
 
   const dispatch = useDispatch();
 
   const dispatchFetchAction = () => {
-    dispatch(setMaterialsLoadingStatus(direction));
+    dispatch(
+      setMaterialsLoadingStatus({
```

Comment:
pls put it to `fetchMaterials` so that fetchMaterials will handle loading status each time.
you are doing the same for initial loading